src/qibolab/instruments/zhinst.py

```diff
@@ -505,14 +503,21 @@ def play(self, qubits, sequence, options):
                 continue
             q = qubit.name  # pylint: disable=C0103
             if len(self.sequence[f"readout{q}"]) != 0:
-                exp_res = self.results.get_data(f"sequence{q}")
-                if options.acquisition_type is AcquisitionType.DISCRIMINATION:
-                    data = np.array([exp_res]) if options.averaging_mode is AveragingMode.CYCLIC else np.array(exp_res)
-                    results[self.sequence[f"readout{q}"][0].pulse.serial] = options.results_type(data)
-                    results[self.sequence[f"readout{q}"][0].pulse.qubit] = options.results_type(data)
-                else:
-                    results[self.sequence[f"readout{q}"][0].pulse.serial] = options.results_type(data=np.array(exp_res))
-                    results[self.sequence[f"readout{q}"][0].pulse.qubit] = options.results_type(data=np.array(exp_res))
+                for i in range(len(self.sequence[f"readout{q}"])):
```

Comment:
I think this for would look a bit better with 
```suggestion
                for i, pulse in enumerate(self.sequence[f"readout{q}"]):
```
so that `self.sequence[f"readout{q}"][i]` => `ropulse`